api.php

```diff
@@ -22,48 +25,58 @@
         $data = array_merge($data,  $sum);
     }
 
-    // Works without authorization
     if (isset($_GET['overTimeData'])) {
         $data = array_merge($data,  getOverTimeData());
     }
 
-    // Works without authorization
     if (isset($_GET['overTimeData10mins'])) {
         $data = array_merge($data,  getOverTimeData10mins());
     }
 
-    // Requires authorization
+    // Auth Required
+
     if (isset($_GET['topItems']) && $auth) {
         $data = array_merge($data,  getTopItems());
     }
 
-    // Requires authorization
     if (isset($_GET['recentItems']) && $auth) {
         if (is_numeric($_GET['recentItems'])) {
             $data = array_merge($data,  getRecentItems($_GET['recentItems']));
         }
     }
 
-    // Requires authorization
     if (isset($_GET['getQueryTypes']) && $auth) {
         $data = array_merge($data, getIpvType());
     }
 
-    // Requires authorization
     if (isset($_GET['getForwardDestinations']) && $auth) {
         $data = array_merge($data, getForwardDestinations());
     }
 
-    // Requires authorization
     if (isset($_GET['getQuerySources']) && $auth) {
         $data = array_merge($data, getQuerySources());
     }
 
-    // Requires authorization
     if (isset($_GET['getAllQueries']) && $auth) {
         $data = array_merge($data, getAllQueries());
     }
 
+    if (isset($_GET['enable'], $_GET['token']) && $auth) {
+        check_csrf($_GET['token']);
+        exec('sudo pihole enable');
+        $data = array_merge($data, Array(
+            "status" => "enabled"
+        ));
+    }
+
+    if (isset($_GET['disable'], $_GET['token']) && $auth) {
```

Comment:
`api.php?enable&disable` would be possible and first enable and then disable the Pi-hole. I think an if ... else ... construct would be more suitable here.